src/reanimated2/animation/spring.ts

```diff
@@ -144,12 +148,35 @@ export function withSpring(
       previousAnimation: SpringAnimation
     ): void {
       animation.current = value;
+
+      let frameRatio;
+
+      if (config.duration) {
+        const c = config.damping;
+        const m = config.mass;
+        const amplitude = Number(animation.toValue) - animation.startValue;
+
+        const expectedDuration =
```

Comment:
I think that we should add link to paper here. Link: https://phys.libretexts.org/Bookshelves/University_Physics/Book%3A_University_Physics_(OpenStax)/Book%3A_University_Physics_I_-_Mechanics_Sound_Oscillations_and_Waves_(OpenStax)/15%3A_Oscillations/15.06%3A_Damped_Oscillations